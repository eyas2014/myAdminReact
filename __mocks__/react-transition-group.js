import React from 'react';

let result=[]
function isCalledWith(){

	return result;
};

function reset(){

	result=[];
};

function CSSTransition(props){
	const {classNames, timeout, unmountOnExit}=props;
	result.push({in: props.in, classNames, timeout, unmountOnExit});
	return (props.children);
}



export {CSSTransition, isCalledWith, reset};