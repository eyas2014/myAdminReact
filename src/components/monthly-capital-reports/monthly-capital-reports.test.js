import React from 'react';
import MonthlyCapReports from './monthly-capital-reports.jsx';
import enzyme from 'enzyme';

jest.mock('react-transition-group');
jest.mock('react-chartjs-2');
jest.mock('./progress.jsx');
jest.mock('./stat-panel.jsx');

document.body.innerHTML='<div  id="root"></div>';

  var salesChartData = 'salesChartData';
  var salesChartOptions={
				scales: {
	    			xAxes: [{
	                	gridLines: {
	                    	display:false
	                	}
	            	}],
	    			yAxes: [{
	                	gridLines: {
	                    	display:false
	                	}   
	            	}]
	    		}
			};


	var progressData=[{
				step: 'Add Products To Cart',
				sum: 200,
				completed: 110
				},{
				step: 'Complete Purchase',
				sum: 400,
				completed: 80
				},{
				step: 'Visit Premium Page',
				sum: 800,
				completed: 700
				},{
				step: 'send Inquiries',
				sum: 500,
				completed: 150 
				}];


const mountWrapper=enzyme.mount(<MonthlyCapReports data={{salesChartData, progressData}}></MonthlyCapReports>, {attachTo: document.getElementById('root')});

it('should render correct htmls',()=>{
	expect(mountWrapper.html()).toMatchSnapshot();
});

it('chartjs should be called with correct arguments', ()=>{
	expect(require('react-chartjs-2').isCalledWith()).toEqual({data: 'salesChartData', options: salesChartOptions})
});

it('should render correct number of progress bar', ()=>{
	expect(require('./progress.jsx').isCalledWith().length).toEqual(4);	
});

it('should render correct number of stat panels', ()=>{
	expect(require('./stat-panel.jsx').isCalledWith().length).toEqual(4);	
});

it('should collapse after clicking toggle', ()=>{
	require('react-transition-group').reset();
	mountWrapper.find('.monthly-capital-reports header').childAt(1).childAt(1).simulate('click');
	expect(require('react-transition-group').isCalledWith()[0].classNames).toEqual('componentShutdown');
	expect(require('react-transition-group').isCalledWith()[0].in).toEqual(true);
	expect(require('react-transition-group').isCalledWith()[1].classNames).toEqual('componentCollapse');
	expect(require('react-transition-group').isCalledWith()[1].in).toEqual(false);
});

it('should expand after clicking toggle', ()=>{
	require('react-transition-group').reset();
	mountWrapper.find('.monthly-capital-reports header').childAt(1).childAt(1).simulate('click');
	expect(require('react-transition-group').isCalledWith()[0].classNames).toEqual('componentShutdown');
	expect(require('react-transition-group').isCalledWith()[0].in).toEqual(true);
	expect(require('react-transition-group').isCalledWith()[1].classNames).toEqual('componentCollapse');
	expect(require('react-transition-group').isCalledWith()[1].in).toEqual(true);
});

it('should close after clicking close', ()=>{
	require('react-transition-group').reset();
	mountWrapper.find('.monthly-capital-reports header').childAt(1).childAt(0).simulate('click');
	expect(require('react-transition-group').isCalledWith()[0].classNames).toEqual('componentShutdown');
	expect(require('react-transition-group').isCalledWith()[0].in).toEqual(false);
});



