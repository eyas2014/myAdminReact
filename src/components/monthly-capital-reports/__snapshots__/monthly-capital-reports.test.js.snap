// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct htmls 1`] = `"<div class=\\"monthly-capital-reports bg-white text-black box-shadow\\"><header class=\\"bg-white box-border-bottom\\"><div class=\\"btn\\"><span>Monthly Recap Report</span></div><div class=\\"btn-group float-right text-muted\\"><div class=\\"btn\\"><i class=\\"fa fa-times\\"></i></div><div class=\\"btn\\"><i class=\\"fa fa-minus\\"></i></div><div class=\\"btn\\"><i class=\\"fa fa-bars\\"></i></div></div><div></div></header><div class=\\"wrapper\\"><main class=\\"box-border-bottom row mx-0\\"><div class=\\"col-8 mx-0 p-3\\"><div id=\\"line-chart\\"></div></div><div class=\\"col-4 p-3\\"><div class=\\"pt-3 text-center\\"><span class=\\"text-center font-weight-bold\\">Goal Completion</span></div><div class=\\"progress\\"></div><div class=\\"progress\\"></div><div class=\\"progress\\"></div><div class=\\"progress\\"></div></div></main><footer class=\\"container-fluid bg-white\\"><div class=\\"row py-4\\"><div class=\\"stat-panel\\"></div><div class=\\"stat-panel\\"></div><div class=\\"stat-panel\\"></div><div class=\\"stat-panel\\"></div></div></footer></div></div>"`;
